src/lib/server/reservation.ts

```diff
@@ -47,44 +63,138 @@ export async function getReservationsCsv(branch: string) {
 		records.map((ent) => {
 			return {
 				..._.omit(ent, ['user']),
-				name: ent.user?.name || ent.user.id,
-				nickname: ent.user?.nickname || 'no nickname'
+				name: ent.user.name,
+				nickname: ent.user.nickname
 			};
 		})
 	);
 	return await csv.toString();
 }
 
+export async function convertFromXataToAppType(rawRsvs: ReservationsRecord[]) {
+	await initSettings();
+
+	// make sure it's safe to cast to the Reservation type
```

Comment:
@getneil here's the code that ensures that reservation entries coming from Xata don't have unexpected null/invalid values